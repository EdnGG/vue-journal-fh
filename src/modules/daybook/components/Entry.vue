<template>
  <div class="entry-container mb-3 pointer p-2"
    @click="$router.push({name: 'entry', params: {id: 10}})">
    <!-- title -->
    <div class="entry-title d-flex">
      <span class="text-success fs-5 fw-bold">15</span>
      <span class="mx-1 fs-5">July</span>
      <span class="mx-2 fw-ligth">2021, Thursday</span>
    </div>
    <div class="entry-description">
    Lorem ipsum dolor sit amet consectetur, adipisicing elit. Libero veritatis corporis esse, sunt, commodi amet dicta odio similique accusamus quas maxime consectetur animi! Dicta eum eos officiis ullam ad animi.
    </div>
  </div>
</template>

<script>
export default {};
</script>

<style lang="scss" scoped>
.entry-container {
  border-bottom: 1px solid #2c3e50;
  transition: 0.2s ease-in all;
  &:hover {
    background-color: lighten($color: grey, $amount: 45);
    transition: 0.2s ease-in all;
  }
  .entry-description {
    font-size: 12px;
  }
}
</style>
