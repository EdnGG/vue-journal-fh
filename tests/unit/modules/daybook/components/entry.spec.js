import { shallowMount } from "@vue/test-utils";
import Entry from "@/modules/daybook/components/Entry";
import { journalState } from "../../../mock-data/test-journal-state";

describe("Testing Entry Component", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: journalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  test("must match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("must redirect when click on entry-container", () => {
    wrapper.find(".entry-container").trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: "entry",
      params: {
        //  id: wrapper.props().entry.id,
        //   id: "-NT6lYa-7HVHjuCGUCmk",
        id: journalState.entries[0].id,
      },
    });
  });

  test("Testing computed properties", () => {
    expect(wrapper.vm.day).toBe("Saturday");
    expect(wrapper.vm.month).toBe("April");
    expect(wrapper.vm.yearDay).toBe("2023, Saturday");
  });
});
