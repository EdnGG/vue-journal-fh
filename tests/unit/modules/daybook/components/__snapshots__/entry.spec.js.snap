// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Entry Component must match the snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- title -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">Saturday</span><span class="mx-1 fs-5">April</span><span class="mx-2 fw-ligth">2023, Saturday</span></div>
  <div class="entry-description">hello world with image
  </div>
</div>
`;
